cfme/tests/ansible/test_embedded_ansible_basic.py

```diff
@@ -214,6 +215,56 @@ def test_embedded_ansible_repository_invalid_url_crud(request, appliance, wait_f
     repository.delete_if_exists()
 
 
+@pytest.mark.rhel_testing
+@pytest.mark.tier(1)
+def test_embedded_ansible_add_private_repository_crud(
+    request, credentials_collection, appliance, wait_for_ansible
+):
+    """
+    Add Private Repository by using SCM credentials. Check repository gets added successfully.
+
+    Polarion:
+        assignee: sbulage
+        casecomponent: Ansible
+        caseimportance: critical
+        initialEstimate: 1/6h
+        tags: ansible_embed
+    """
+    # Create SCM credentials.
+    creds = {
+        "username": conf.credentials["scm_creds"]["username"],
+        "password": conf.credentials["scm_creds"]["password"],
+    }
+
+    scm_credential = credentials_collection.create(
+        "Scm_credential",
+        "Scm",
+        **creds
+    )
+
+    assert scm_credential.exists
+
+    # Create private repository using SCM credentials.
+    repositories = appliance.collections.ansible_repositories
+    repository = repositories.create(
+        name=fauxfactory.gen_alpha(),
+        url="https://gitlab.com/cfme/ansible_private_repo.git",
+        description=fauxfactory.gen_alpha(),
+        scm_credentials=scm_credential.name,
+    )
+    view = navigate_to(repository, "Details")
+    wait_for(
+        lambda: repository.status == "successful",
+        num_sec=120,
+        delay=2,
+        fail_func=view.toolbar.refresh.click
+    )
+    assert repository.exists
+
+    scm_credential.delete_if_exists()
+    repository.delete_if_exists()
```

Comment:
As its crud operation better we `delete` and check check  for `exists`.
`delete_if_exists` generally we use for teardown means delete entity if exist but in this case we should assert not exists.